reference/hoon-expressions/rune/lus.md

```diff
@@ -135,42 +135,74 @@ Any Hoon expression, `q`, may be used to define the arm computation.
 
 ### +* "lustar"
 
-
-Produce a type constructor arm.
+Used to assign an alias to doors and define virtual arms.
 
 ##### Syntax
 
-Regular: **2-fixed**.
+Regular: **variadic**.
 
 ```hoon
-+*  p=term  [q=term]  r=spec
++*  a=term  b=hoon
+    c=term  d=hoon
+    ...
+    e=term  f=hoon
 ```
 
-`p` is the arm name, `q` is the name of the argument for the constructor function enclosed in square brackets, and `r` is a structure expression that defines the constructor function.
+`a`, `c`, `e` are arm names and `b`, `d`, `f` are any Hoon expression. Note that unlike all other
+runes with a variable number of arguments, the list of arguments of `+*` does
+not end with a terminating rune.
+
+`+*` arms must always come at the beginning of the battery, before any other
+type of lus arm.
 
 ##### Discussion
 
-Arms produced by `+*` are essentially type constructors.  They are used to construct new types from ones you already have.  Consider `list`, which takes some type -- e.g., `@`, `tape`, etc. -- and returns a new type, e.g., `(list @)`, `(list tape)`, etc.
+The primary use of `+*` is to assign aliases to doors (see Examples below).
+However, they may also be used to define "virtual arms" which behave just like
+ordinary arms except they allow nesting within cores that requires a specific
+number of arms, such as Gall apps. Put another way, no matter how many virtual
+arms are defined in a `+*` statement, the interpreter will see it as only being
+a single arm for the purposes of nesting.
 
-The Hoon subexpression, `r`, must be a structure expression.  That is, it must be either a basic structure expression (`*`, `~`, `^`, `?`, and `@`), or a complex expression made with the `$` family of runes (including irregular variants).
+Under the hood, `+*` gets compiled to `=*`.
```

Comment:
See [its destructuring](https://github.com/urbit/urbit/blob/3196179fbdc7afe4f5d4c8143b85ee7c5b9c7d41/pkg/arvo/sys/hoon.hoon#L8680-L8688).

If I'm getting this right (@joemfb please confirm) that means that this:

```hoon
|_  z=@ud
+*  n  1
++  x  (add z n)
++  y  (sub z n)
--
```

becomes roughly:

```hoon
=|  z=@ud
|%
++  x
  =*  n  1
  (add z n)
++  y
  =*  n  1
  (sub z n)
--
```

Reply:
That's right.